code/modules/mob/living/basic/vermin/butterfly.dm

```diff
@@ -44,3 +44,55 @@
 
 /mob/living/basic/butterfly/lavaland
 	unsuitable_atmos_damage = 0
+
+/mob/living/basic/butterfly/lavaland/temporary
+	name = "strange butterfly"
+	var/obj/item/mod/core/plasma/lavaland/source = NONE
+	// Max distance in tiles before despawning
+	var/max_distance = 5
+	var/will_be_destroyed = FALSE
+	var/despawn_timer = 0
+
+/mob/living/basic/butterfly/lavaland/temporary/Initialize(mapload, creator)
+	. = ..()
+	source = creator
+	START_PROCESSING(SSprocessing, src)
+
+/mob/living/basic/butterfly/lavaland/temporary/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	if(source)
+		source.child_despawned()
+		source.children -= src
+	. = ..()
+
+/mob/living/basic/butterfly/lavaland/temporary/process()
+	if(should_despawn())
+		will_be_destroyed = TRUE
+		despawn_timer = addtimer(CALLBACK(src, TYPE_PROC_REF(/mob/living/basic/butterfly/lavaland/temporary, fadeout)), 5 SECONDS, TIMER_STOPPABLE)
+	else
+		if(will_be_destroyed)
+			// Cancels the butterfly being destroyed
+			will_be_destroyed = FALSE
+			deltimer(despawn_timer)
+
+/mob/living/basic/butterfly/lavaland/temporary/proc/should_despawn()
+	if(!source.mod.active)
```

Comment:
This should also use a signal, register to `COMSIG_MODULE_DEACTIVATED` and trigger the deaths of all butterflies at that point.